tests/rados/test_bluestoretool_workflows.py

```diff
@@ -0,0 +1,272 @@
+            )
+            lvm_list, _ = osd_node.exec_command(
+                sudo=True,
+                cmd=f"cephadm shell -- ceph-volume lvm list {osd_id} --format json",
+            )
+            lvm_json = json.loads(lvm_list)
+            dev = lvm_json[f"{osd_id}"][0]["tags"]["ceph.block_device"]
+            device_type = lvm_json[f"{osd_id}"][0]["type"]
+            if device in device_type:
+                log.info(
+                    f"\n --------------------"
+                    f"\n Dump label content for {device_type} device {dev} for OSD {osd_id}"
+                    f"\n --------------------"
+                )
+                out = bluestore_obj.show_label(osd_id=osd_id, device=dev)
+                log.info(out)
+                assert dev in out
+
+        # Execute ceph-bluestore-tool prime-osd-dir --dev <main_device> --path <osd_path>
+        osd_id = random.randrange(len(osd_list))
+        osd_node = rados_obj.fetch_host_node(daemon_type="osd", daemon_id=str(osd_id))
+        lvm_list, _ = osd_node.exec_command(
+            sudo=True,
+            cmd=f"cephadm shell -- ceph-volume lvm list {osd_id} --format json",
+        )
+        lvm_json = json.loads(lvm_list)
+        dev = lvm_json[f"{osd_id}"][0]["tags"]["ceph.block_device"]
+        log.info(
+            f"\n --------------------"
+            f"\n Generate the content for an OSD data directory "
+            f"that can start up a BlueStore OSD for OSD {osd_id}"
+            f"\n --------------------"
+        )
+        out = bluestore_obj.generate_prime_osd_dir(osd_id=osd_id, device=dev)
+        log.info(out)
+
+        # Execute ceph-bluestore-tool free-dump --path <osd_path> [--allocator block/bluefs-wal/bluefs-db/bluefs-slow]
+        osd_id = random.randrange(len(osd_list))
+        log.info(
+            f"\n --------------------"
+            f"\n Dump all free regions in allocator for OSD {osd_id}"
+            f"\n --------------------"
+        )
+        out = (
+            bluestore_obj.get_free_dump(osd_id=osd_id)
+            if rhbuild.startswith("6")
+            else bluestore_obj.get_free_dump(osd_id=osd_id, allocator_type="block")
+        )
+        log.debug(out)
+        assert "alloc_name" in out and "extents" in out
+
+        # Execute ceph-bluestore-tool free-score --path <osd_path> [--allocator block/bluefs-wal/bluefs-db/bluefs-slow]
+        osd_id = random.randrange(len(osd_list))
+        log.info(
+            f"\n --------------------"
+            f"\n Get fragmentation score for OSD {osd_id}"
+            f"\n --------------------"
+        )
+        out = (
+            bluestore_obj.get_free_score(osd_id=osd_id)
+            if rhbuild.startswith("6")
+            else bluestore_obj.get_free_score(osd_id=osd_id, allocator_type="block")
+        )
+        log.info(out)
+        assert "fragmentation_rating" in out
+
+        # Execute ceph-bluestore-tool show-sharding --path <osd_path>
```

Comment:
Resharding flow is not tested. Please add a comment to consolidate commands which are not exercised in this PR.

Reply:
Included in the test docstring line 34